src/NumSharp.Core/Backends/Iterators/NDIterator.cs

```diff
@@ -128,7 +128,7 @@ protected void setDefaults_NoCast()
                     case IteratorType.Scalar:
                     {
                         var hasNext = new Reference<bool>(true);
-                        var offset = shape.TransformOffset(0);
+                        var offset = shape.GetOffset(0);
```

Comment:
Revert this please, it has to be transform - the shape's dim might not be 1 and you are passing in new int[] {0}

Reply:
done.